credsweeper/app.py

```diff
@@ -177,42 +198,34 @@ def __multi_jobs_scan(self, content_providers: Union[List[DiffContentProvider],
                 pool.join()
                 sys.exit()
 
-    def file_scan(self, file_provider: ContentProvider) -> List[Candidate]:
+    # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+
+    def file_scan(self, content_provider: ContentProvider) -> List[Candidate]:
         """Run scanning of file from 'file_provider'.
 
         Args:
-            file_provider: file provider object to scan
+            content_provider: content provider object to scan
 
         Return:
             list of credential candidates from scanned file
 
         """
-        # Get list credentials for each file
-        logging.debug(f"Start scan file: {file_provider.file_path}")
-
-        if self.config.find_by_ext:
-            if FilePathExtractor.is_find_by_ext_file(self.config, file_provider.file_path):
-                candidate = Candidate(
-                    line_data_list=[
-                        LineData(  #
-                            self.config,  #
-                            line="dummy line",  #
-                            line_num=-1,  #
-                            path=file_provider.file_path,  #
-                            pattern=regex.compile(".*"))
-                    ],
-                    patterns=[regex.compile(".*")],  #
-                    rule_name="Dummy candidate",  #
-                    severity=Severity.INFO,  #
-                    config=self.config)
-                return [candidate]
-
-        try:
-            scan_context = file_provider.get_analysis_target()
-            return self.scanner.scan(scan_context)
-        except UnicodeDecodeError:
```

Comment:
You have removed exception check here.
Is this done by purpose?

Reply:
it is done inside get_analysis_target and  scan. Moreover, the line was not covered with fuzzing that generated many various bytes that may throw the exception. 
So, the exception is extra here.